packages/ui/src/components/pages/Accounts/AccountControls/RenameAccountDialog.tsx

```diff
@@ -0,0 +1,83 @@
+import { ChangeEvent, FormEvent, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { toast } from 'react-toastify';
+import { useEffectOnce } from 'react-use';
+import { Button, Dialog, DialogContent, DialogContentText, TextField } from '@mui/material';
+import DialogTitle from 'components/shared/DialogTitle';
+import EmptySpace from 'components/shared/misc/EmptySpace';
+import useDialog from 'hooks/useDialog';
+import { useWalletState } from 'providers/WalletStateProvider';
+import { AccountInfoExt } from 'types';
+import { EventName, EventRegistry } from 'utils/eventemitter';
+
+export default function RenameAccountDialog(): JSX.Element {
+  const { keyring } = useWalletState();
+  const { t } = useTranslation();
+  const { open, doOpen, doClose } = useDialog();
+  const [name, setName] = useState<string>('');
+  const [account, setAccount] = useState<AccountInfoExt>();
+
+  const isInvalidName = useMemo(() => name.length > 16, [name]);
```

Comment:
Should the below be sufficient instead of using a `useMemo`?
```suggestion
  const isInvalidName = name.length > 16;
```